src/dxtbx/datablock.py

```diff
@@ -759,11 +754,9 @@ def from_filenames(
 
         # Datablocks can only contain imageset with a shared format class, therefore
         # group the experiments by format class
-        format_groups = collections.OrderedDict()
+        format_groups = collections.defaultdict(ExperimentList)
         for expt in expts:
-            format_class = expt.imageset.get_format_class()
-            format_groups.setdefault(format_class, ExperimentList())
-            format_groups[format_class].append(expt)
+            format_groups[expt.imageset.get_format_class()].append(expt)
```

Comment:
just checking - it is known that this is not equivalent, right?
(The new code allows access to all undeclared keys)

Reply:
Yes, but given that this is only a temporary dictionary that is internal to this method, I thought it was safe and worth the convenience of not having to explicitly call `.setdefault` to create an empty `ExperimentList`. Happy to stick to standard Python `dict` if you think that is preferable.

Reply:
Discussed and not really any more confusing than what was there before.